src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php

```diff
@@ -11,68 +11,224 @@
+            }
+            $options += static::$defaultClientOptions;
+            $client = new \Memcached($options['persistent_id']);
+            $username = $options['username'];
+            $password = $options['password'];
+            unset($options['persistent_id'], $options['username'], $options['password']);
+
+            // set client's options
+            foreach ($options as $name => $value) {
+                if (is_int($name)) {
+                    continue;
+                }
+                $ucName = strtoupper($name);
+                $opt = constant('Memcached::OPT_'.$ucName);
+
+                if ('HASH' === $ucName || 'SERIALIZER' === $ucName || 'DISTRIBUTION' === $ucName) {
+                    $value = constant('Memcached::'.$ucName.'_'.strtoupper($value));
+                }
+
+                unset($options[$name]);
+                $options[$opt] = $value;
+            }
+            $client->setOptions($options);
+
+            // parse any DSN in $servers
+            foreach ($servers as $i => $dsn) {
+                if (is_array($dsn)) {
+                    continue;
+                }
+                if (0 !== strpos($dsn, 'memcached://')) {
+                    throw new InvalidArgumentException(sprintf('Invalid Memcached DSN: %s does not start with "memcached://"', $dsn));
+                }
+                $params = preg_replace_callback('#^memcached://(?:([^@]*+)@)?#', function ($m) use (&$username, &$password) {
+                    if (!empty($m[1])) {
+                        list($username, $password) = explode(':', $m[1], 2) + array(1 => null);
+                    }
+
+                    return 'file://';
+                }, $dsn);
+                if (false === $params = parse_url($params)) {
+                    throw new InvalidArgumentException(sprintf('Invalid Memcached DSN: %s', $dsn));
+                }
+                if (!isset($params['host']) && !isset($params['path'])) {
+                    throw new InvalidArgumentException(sprintf('Invalid Memcached DSN: %s', $dsn));
+                }
+                if (isset($params['path']) && preg_match('#/(\d+)$#', $params['path'], $m)) {
+                    $params['weight'] = $m[1];
+                    $params['path'] = substr($params['path'], 0, -strlen($m[0]));
+                }
+                $params += array(
+                    'host' => isset($params['host']) ? $params['host'] : $params['path'],
+                    'port' => isset($params['host']) ? 11211 : null,
+                    'weight' => 0,
+                );
+                if (isset($params['query'])) {
+                    parse_str($params['query'], $query);
+                    $params += $query;
+                }
+
+                $servers[$i] = array($params['host'], $params['port'], $params['weight']);
+            }
+
+            // set client's servers, taking care of persistent connections
+            if (!$client->isPristine()) {
```

Comment:
Not sure to get what you mean. Do you see anything wrong with this logic?
Here it is: if the connection is not pristine (meaning its a 2nd use persistent connection), then we check if there is any change in the list of server. If there is, we reset the list and set the new one.